shell/platform/android/io/flutter/embedding/engine/android/AndroidTouchProcessor.java

```diff
@@ -57,27 +79,43 @@ public boolean onTouchEvent(MotionEvent event) {
 
     // Prepare a data packet of the appropriate size and order.
     ByteBuffer packet = ByteBuffer.allocateDirect(
-        pointerCount * POINTER_DATA_FIELD_COUNT * BYTE_PER_FIELD
+        pointerCount * POINTER_DATA_FIELD_COUNT * BYTES_PER_FIELD
     );
     packet.order(ByteOrder.LITTLE_ENDIAN);
 
     int maskedAction = event.getActionMasked();
-    // ACTION_UP, ACTION_POINTER_UP, ACTION_DOWN, and ACTION_POINTER_DOWN
-    // only apply to a single pointer, other events apply to all pointers.
-    if (maskedAction == MotionEvent.ACTION_UP || maskedAction == MotionEvent.ACTION_POINTER_UP
-        || maskedAction == MotionEvent.ACTION_DOWN || maskedAction == MotionEvent.ACTION_POINTER_DOWN) {
-      addPointerForIndex(event, event.getActionIndex(), packet);
+    int pointerChange = getPointerChangeForAction(event.getActionMasked());
+    if (maskedAction == MotionEvent.ACTION_DOWN || maskedAction == MotionEvent.ACTION_POINTER_DOWN) {
+      // ACTION_DOWN and ACTION_POINTER_DOWN always apply to a single pointer only.
```

Comment:
subjective, optional: Named conditional variables would remove or at least reduce the need for implementation comments here. It's debatable but in my experience named variables stay more in sync with the code over time than comments.

```java
boolean updateForSinglePointer = maskedAction == MotionEvent.ACTION_DOWN || maskedAction == MotionEvent.ACTION_POINTER_DOWN;
boolean updateForMultiplePointers = !updateForSinglePointer && (maskedAction == MotionEvent.ACTION_UP || maskedAction == MotionEvent.ACTION_POINTER_UP);
if (updateForSinglePointer) {
   // ...
```

Reply:
Good point.  Done.